src/test/java/uk/gov/digital/ho/pttg/api/AuditHistoryResourceRecordsIntTest.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.digital.ho.pttg.api;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.JsonPath;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlGroup;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.AFTER_TEST_METHOD;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.BEFORE_TEST_METHOD;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@SqlGroup({
+        @Sql(executionPhase = BEFORE_TEST_METHOD, scripts = "/sql/AuditHistoryRecordsIntTest/before.sql"),
+        @Sql(executionPhase = AFTER_TEST_METHOD, scripts = "/sql/AuditHistoryRecordsIntTest/after.sql")
+})
+public class AuditHistoryResourceRecordsIntTest {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    public void shouldRetrieveRecords() throws IOException {
+        String url = "/records?correlationId=correlation-id-1&eventTypes=INCOME_PROVING_FINANCIAL_STATUS_REQUEST,INCOME_PROVING_FINANCIAL_STATUS_RESPONSE";
+        String response = restTemplate.getForObject(url, String.class);
+
+        assertThat(objectMapper.readValue(response, List.class))
```

Comment:
I think using JsonPath is a better way to assert that the actual data provided by a method is correct - without unmarshalling it first. So is it possible to check that the response has two correlation Ids using JsonPath (rather than unmarshalling it)?

Reply:
I have used JsonPath for this purpose on lines 44 and 45. I just used the objectMapper to check that the number of records is as expected as I don't know a way to do that with JsonPath